components/locid/src/subtags/variants.rs

```diff
@@ -46,6 +46,26 @@ impl Variants {
         Self(ShortVec::new())
     }
 
+    /// Creates a new [`Variants`] set from a [`Option<Variant>`].
+    /// The caller is expected to provide a single variant or None.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use icu::locid::subtags::{Variant, Variants};
+    ///
+    /// let variant: Variant = "posix".parse().expect("Parsing failed.");
+    /// let mut v = Some(variant);
+    /// let variants = Variants::from_optional_variant(v);
+    /// ```
+    #[inline]
+    pub const fn from_optional_variant(variant: Option<Variant>) -> Self {
```

Comment:
@zbraniecki Does the name of this function make sense to you?

Reply:
I also think `from_optional_variant` is not the best. Happy to revise if anyone has a better idea.

Reply:
Yeah, I don't like this method too much, esp as public.

Can we instead remove it and on the callee side do:

```
variants: variant.map(|v| Variants::from(v)).unwrap_or_default()
```

Reply:
So one issue is that `Variants::from(v)` is not `const`.

What we could do is make `pub const fn Variants::from_single_variant(v: Variant)`.

Reply:
Why not `from_variant`?

Reply:
WFM

Reply:
@sffc @zbraniecki Two issues when I attempt to apply this change:
1. The trait bound  ~const FnOnce<(Variant,)> is not satisfied.
2. The trait bound Variants: ~const Default is not satisfied.
Is it possible to define const traits?

Reply:
Done